pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10954531?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10954531?buildTab=artifacts#/) on release-23.1 @ [ae14f647a4bda2532bc9f2be41f25f6acc113412](https://github.com/cockroachdb/cockroach/commits/ae14f647a4bda2532bc9f2be41f25f6acc113412):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1099239144
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1099239144
--- FAIL: TestSQLStatsCompactor (181.94s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 33
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1
        	Messages:   	expected 32 number of wide scans issued, but 33 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1 (60.23s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29872

--- a/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
@@ -141,7 +141,8 @@ func TestSQLStatsCompactor(t *testing.T) {
 							},
 						},
 						Store: &kvserver.StoreTestingKnobs{
-							TestingRequestFilter: kvInterceptor.intercept,
+							TestingRequestFilter:      kvInterceptor.intercept,
+							DisableLoadBasedSplitting: true,
 						},
 					},
 				},

